openslides_backend/presenter/check_mediafile_id.py

```diff
@@ -38,30 +43,71 @@ class CheckMediafileId(BasePresenter):
     schema = check_mediafile_id_schema
 
     def get_result(self) -> Any:
-        mediafile = self.datastore.get(
-            FullQualifiedId(Mediafile.collection, self.data["mediafile_id"]),
-            mapped_fields=["filename", "is_directory"],
-        )
+        try:
+            mediafile = self.datastore.get(
+                FullQualifiedId(Mediafile.collection, self.data["mediafile_id"]),
+                mapped_fields=[
+                    "filename",
+                    "is_directory",
+                    "owner_id",
+                    "token",
+                    "mimetype",
+                ],
+            )
+        except DatastoreException:
+            return {"ok": False}
+        if not mediafile.get("owner_id") or mediafile.get("is_directory"):
+            return {"ok": False}
+        self.check_permissions()
+        collection, _ = mediafile["owner_id"].split(KEYSEPARATOR)
+        if collection == "organization":
+            return self.get_organization_result(mediafile)
+        elif collection == "meeting":
+            return self.get_meeting_result(mediafile)
+        return {"ok": False}
 
-        if not mediafile or mediafile.get("is_directory"):
+    def get_organization_result(self, mediafile: Dict[str, Any]) -> Any:
+        if not mediafile.get("token"):
+            return {"ok": True, "filename": mediafile.get("filename")}
+        if not mediafile.get("mimetype"):
             return {"ok": False}
         self.check_permissions()
+        extension = mimetypes.guess_extension(mediafile["mimetype"])
+        if extension is None:
+            return {"ok": False}
+        filename = mediafile["token"] + extension
+
+        return {"ok": True, "filename": filename}
 
+    def get_meeting_result(self, mediafile: Dict[str, Any]) -> Any:
+        if not mediafile.get("filename"):
+            return {"ok": False}
         return {"ok": True, "filename": mediafile["filename"]}
 
     def check_permissions(self) -> None:
         mediafile = self.datastore.get(
             FullQualifiedId(Mediafile.collection, self.data["mediafile_id"]),
             [
-                "meeting_id",
+                "owner_id",
                 "used_as_logo_$_in_meeting_id",
                 "used_as_font_$_in_meeting_id",
                 "projection_ids",
                 "is_public",
                 "inherited_access_group_ids",
+                "token",
             ],
         )
-        meeting_id = mediafile.get("meeting_id")
+
+        # Try to get the meeting id.
+        meeting_id = None
+        if mediafile.get("owner_id"):
+            if mediafile["owner_id"].split(KEYSEPARATOR)[0] == "organization":
```

Comment:
save result of `split` for easier access

Reply:
Improved the handling of the owner_collection and the owner_id_int.
The name owner_id_int is strange, but I think owner_id_id is worse.

Reply:
Okay, I have renamed `owner_id_int` into `owner_id`.